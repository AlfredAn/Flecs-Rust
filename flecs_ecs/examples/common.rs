#![allow(dead_code)]

pub use flecs_ecs::{core::*, macros::Component};

#[derive(Debug, Default, Clone, Component)]
pub struct Position {
    pub x: f32,
    pub y: f32,
}

#[derive(Default, Debug, Clone, Component)]
pub struct Velocity {
    pub x: f32,
    pub y: f32,
}

#[derive(Default, Clone, Component)]
pub struct Mass {
    pub value: f32,
}

#[derive(Default, Clone, Component)]
pub struct Eats;

#[derive(Default, Clone, Component)]
pub struct Apples;

#[derive(Default, Clone, Component)]
pub struct Pears;

#[derive(Default, Clone, Component)]
pub struct Walking;

#[derive(Default, Clone, Component)]
pub struct Tag;

#[derive(Default, Clone, Component)]
pub struct Human;

#[derive(Default, Clone, Component)]
pub struct Attack {
    pub value: f32,
}

#[derive(Default, Clone, Component)]
pub struct Defence {
    pub value: f32,
}

#[derive(Default, Clone, Component)]
pub struct Damage {
    pub value: f32,
}

#[derive(Default, Clone, Component)]
pub struct FreightCapacity {
    pub value: f32,
}

#[derive(Default, Clone, Component)]
pub struct ImpulseSpeed {
    pub value: f32,
}

#[derive(Default, Clone, Component)]
pub struct HasFlt;

#[derive(Default, Clone, Component)]
pub struct First;

#[derive(Default, Clone, Component)]
pub struct Second;

#[derive(Default, Clone, Component)]
pub struct Third;

#[derive(Default, Clone, Component)]
pub struct Group;

#[allow(dead_code)]
fn main() {
    //this file is for common structs and functions used in the examples
}
