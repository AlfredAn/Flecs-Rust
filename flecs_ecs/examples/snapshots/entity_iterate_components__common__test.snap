---
source: flecs_ecs/examples/common.rs
expression: self.str
---
- "Bob's components:"
- "[entity_iterate_components.common.Position, entity_iterate_components.common.Velocity, entity_iterate_components.common.Human, (Identifier,Name), (entity_iterate_components.common.Eats,entity_iterate_components.common.Apples)]"
- "\n"
- "0: entity_iterate_components.common.Position"
- "1: entity_iterate_components.common.Velocity"
- "2: entity_iterate_components.common.Human"
- "3: (Identifier,Name)"
- "4: (entity_iterate_components.common.Eats,entity_iterate_components.common.Apples)"
- "\n0: entity: Position1: entity: Velocity2: entity: Human3: rel: Identifier, target: Name4: rel: Eats, target: Apples"
- "Position's components:"
- "[Component, (Identifier,Name), (Identifier,Symbol), (ChildOf,entity_iterate_components.common), (OnDelete,Panic)]"
- "\n"
- "0: Component"
- "1: (Identifier,Name)"
- "2: (Identifier,Symbol)"
- "3: (ChildOf,entity_iterate_components.common)"
- "4: (OnDelete,Panic)"
- "\n0: entity: Component1: rel: Identifier, target: Name2: rel: Identifier, target: Symbol3: rel: ChildOf, target: common4: rel: OnDelete, target: Panic"
